### PR TITLE
Add recursive triggers for web

### PR DESCRIPTION
**Pull request by mugikhan**

### CODE REVIEW

File: packages/powersync/lib/src/web/powersync_db.worker.dart [hunk 1]
```diff
@@ -21,6 +21,7 @@ final class PowerSyncAsyncSqliteController extends AsyncSqliteController {
   Future<WorkerDatabase> openDatabase(
       WasmSqlite3 sqlite3, String path, String vfs) async {
     final db = sqlite3.open(path, vfs: vfs);
+    db.execute('PRAGMA recursive_triggers = TRUE');
```
Comment (stevensJourney) [inline packages/powersync/lib/src/web/powersync_db.worker.dart:24]: This pragma statement is already present [here](https://github.com/powersync-ja/powersync.dart/blob/171f848a983ee304b362ca307235127b1cd7db2e/packages/powersync/lib/src/open_factory/abstract_powersync_open_factory.dart#L26)

Seems like the issue is that the `open` method of `AbstractPowerSyncOpenFactory` executes these pragma statements. The `open` method is used by the `openConnection` method for native implementations, but is not used for web. 

I think we should be able to override the `openConnection` method in [here](https://github.com/powersync-ja/powersync.dart/blob/sqlite_proto/packages/powersync/lib/src/open_factory/web/web_open_factory.dart) To execute relevant pragmas.

File: packages/powersync/lib/src/open_factory/web/web_open_factory.dart [hunk 2]
```diff
@@ -14,6 +17,16 @@ class PowerSyncOpenFactory extends AbstractPowerSyncOpenFactory {
     // No op for web
   }
 
+  @override
+  Future<SqliteConnection> openConnection(SqliteOpenOptions options) async {
+    var conn = await super.openConnection(options);
+    for (final statement in super.pragmaStatements(options)) {
+      conn.execute(statement);
```
Comment (stevensJourney) [inline packages/powersync/lib/src/open_factory/web/web_open_factory.dart:24]: should we await this?

Comment (mugikhan) [reply packages/powersync/lib/src/open_factory/web/web_open_factory.dart]: Yes, `execute` is a future. Fixed it.

**Review by stevensJourney — APPROVED** [review]

**Comment by rkistner** [comment]
Pana checks fail on version bumps - safe to ignore.

**Pull request merged**